storage-s3-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java

```diff
@@ -103,9 +144,12 @@ public void write(
         if (overwrite) {
             writeActions(fs, path, actions);
             return;
+        } else if (fs.exists(path)) {
```

Comment:
Now I think this check is necessary. Without it, the only protection is just coming from DymamoDB. If the entry gets removed from DymamoDB for whatever reason, anyone can overwrite an existing file.

Reply:
I still think this part should be as close as possible to the dynamoDB check, in order to minimize the risky time in which the entry can be removed from ddb.

Reply:
That's a good point. +1

Reply:
> I still think this part belongs as close as possible to the dynamoDB check, in order to minimize the risky time in which the entry can be removed from ddb.

But if we already set the TTL delay was 1 day, I don't see why this code optimization is necessary?

Plus, with what you suggested (moving it closer), then we will still be creating the temp file and an entry in DDB, which is unecessary.

Reply:
Let's leave it as it is. Commit conflicts would be more common than incorrect short TTL delay config.

Reply:
I agree that it’s unnecessary given 1 day ttl, but can we be sure it’s kept?

anyway my suggestion is to place it before the insert to ddb so no extra entry will be entered.
I don’t think creating a temp file is a big deal but you can also place it before it,
just after that part:
`// Step 1: Ensure that N-1.json exists`